diagrams/app_diagrams_generator.rb

```diff
@@ -0,0 +1,219 @@
+require 'yaml'
+require 'fileutils'
+
+class AppDiagramGenerator
+  def initialize
+    @repository_path = Dir.getwd
+    @application_data_load_path = File.join(@repository_path, "data/applications.yml")
+    @application_diagrams_save_path = File.join(@repository_path, "diagrams/")
+
+    @applications =
+      YAML.load_file(@application_data_load_path)
+        .reject { |a| a.fetch("retired", false) }
+
+    @applications_by_type =
+      @applications
+      .sort_by { |a| a["github_repo_name"] }
+      .group_by { |a| a["type"] }
+
+    @application_type =
+      Hash[@applications.collect { |a| [a["github_repo_name"], a["type"]] }]
+
+    @actors_by_application_type =
+      @applications_by_type
+      .map { |type, apps|
+        [
+          type,
+          apps.map { |app| app["actors"] }
+          .flatten
+          .reject(&:nil?)
+          .uniq
+        ]
+      }.to_h
+  end
+
+  def get_valid_alias(title)
+    title.tr(".", "_").tr(" ", "_").tr("-", "_")
```

Comment:
We use activesupport in the docs, so you could replace this with `title.underscore` for this.

Reply:
replaced